includes/class-wc-stripe.php

```diff
@@ -862,23 +863,54 @@ public function initialize_status_page() {
 	}
 
 	/**
-	 * Maybe deactivate Affirm or Klarna payment methods if other official plugins are active.
+	 * Deactivate payment methods that should be deactivated, e.g. unreleased,
+	 * BNPLs when other official plugins are active, etc.
 	 *
 	 * @return void
 	 */
-	public function maybe_deactivate_bnpls() {
-		$has_affirm_plugin_active = WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_AFFIRM );
-		$has_klarna_plugin_active = WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_KLARNA );
-		if ( ! $has_affirm_plugin_active && ! $has_klarna_plugin_active ) {
-			return;
-		}
-
+	public function maybe_deactivate_payment_methods() {
 		$gateway = $this->get_main_stripe_gateway();
 		if ( ! is_a( $gateway, 'WC_Stripe_UPE_Payment_Gateway' ) ) {
 			return;
 		}
 
+		$payment_method_ids_to_disable = [];
 		$enabled_payment_methods       = $gateway->get_upe_enabled_payment_method_ids();
+
+		// Check for BNPLs that should be deactivated.
+		$payment_method_ids_to_disable = array_merge(
```

Comment:
We could remove the first `array_merge` but I think it is better to read this way.